packages/block-library/src/index.js

```diff
@@ -208,7 +210,16 @@ export const __experimentalRegisterExperimentalCoreBlocks =
 								queryLoop,
 								queryPagination,
 								postTitle,
-								postContent,
+								/*
+								 * Only provide the Post Content block when editing templates or,
+								 * template parts so there's no risk of posts including themselves
+								 * through a Post Content block (leading to infinite recursion).
+								 */
+								[ 'wp_template', 'wp_template_part' ].includes(
+									postType
+								)
+									? postContent
```

Comment:
should we exclude more site blocks, like postAuthor or postComments, etc?

Reply:
:thinking: The main rationale for this PR was to prevent blockception (infinite recursion), which wouldn't happen with those blocks. But yeah, we might want to exclude the class of FSE relevant `post...` blocks from the content editor. They used to be kind of handy for testing, but maybe that's no longer needed now that we can do that in the Site Editor.

Reply:
Please no, or at least not yet! 😄 
The site editor is unreliable when it comes to loading post attributes into FSE blocks, while it's very easy to test post blocks in the post editor.
I agree that we should eventually limit FSE blocks to the Site Editor, but let's do it once we make it work appropriately.